chore: Begin introducing patterns for Multichain AssetList

gambinish opened the pull request:
## **Description**

This PR introduces some non-breaking patterns from https://github.com/MetaMask/metamask-extension/pull/28386

The idea is to introduce this as a smaller PR to reduce the diff in https://github.com/MetaMask/metamask-extension/pull/28386 to make it easier to review

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/28429?quickstart=1)

## **Related issues**

Fixes: N/A

## **Manual testing steps**

App should be fully functional with no regressions

## **Screenshots/Recordings**

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

gambinish committed:
chore: Style updates to asset-list-control-bar

gambinish committed:
chore: Update localization string in prep for multichain asset list

gambinish committed:
chore: Update import button to import icon in preparation for Multichain asset list

gambinish committed:
chore: Update sort control styles

gambinish committed:
chore: Consider chainId when navigating from main asset list

gambinish committed:
chore: Begin incorporating concept of chainId to support multichain asset list

gambinish committed:
fix: Tweak unit test

gambinish committed:
fix: Tweak unit test

gambinish committed:
chore: Add low risk selectors to prepare for multichain asset list

gambinish committed:
fix: Add unit testable utility in preperation for unified asset list

gambinish committed:
fix: Revert changes to locales, can't do this in advance

gambinish committed:
fix: Update test id in custom-token-add-approve.spec.js

gambinish committed:
chore: Remove console.log

gambinish committed:
fix: Update token-sort.spec.ts testid

gambinish committed:
fix: Update other instances of testid Import

chloeYue committed:
test: [POM] Migrate autodetect and import nft e2e tests to use Page Object Model (#28383)

## **Description**

- Migrate e2e tests `test/e2e/tests/tokens/nft/auto-detect-nft.spec.js`
to TS and Page Object Model, to reduce flakiness.
- Migrate e2e tests `test/e2e/tests/tokens/nft/import-nft.spec.js` to TS
and Page Object Model, to reduce flakiness.
- Create page classe functions for nft
- Deprecate/remove old functions in `helper.js`


[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27155?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/28388

## **Manual testing steps**
Check code readability, make sure tests pass.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

seaona committed:
test: fix state fixtures race condition (#28421)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**
There is a race condition in our testing setup, which causes that the
expected fixtures state, not being there when we start the test. This
has been surfaced in [this
branch](https://github.com/MetaMask/metamask-extension/pull/28277),
where the account tracker multi polling is being added.
The problem is that if we don't have the AccountTracker in state when
the `resetState` function is called (at the beginning of wallet loading)
the balance will remain loading until we refresh the wallet.

Edit: performing the load state early in the test setup fixes the issue.

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/28421?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/3627

## **Manual testing steps**

1. Check all tests continue to pass
2. Check this changes fix this branch ENS test
https://github.com/MetaMask/metamask-extension/pull/28402/files#diff-1acb7898d60977530c97169551d22dbe477a4e3aeb74f1f14bf2eea0b4d75d35
. Alternatively, see videos below with before and after behaviours

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**




https://github.com/user-attachments/assets/8f50ec04-cf96-478e-9c3c-dce54254a628



### **After**


https://github.com/user-attachments/assets/0f109b1a-9289-48d9-b337-d51890c9d448




## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

montelaidev committed:
feat: btc send flow e2e (#28340)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

This PR reintroduces the happy path send flow tests for BTC

## **Related issues**

Fixes: https://github.com/MetaMask/accounts-planning/issues/668

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: Charly Chevalier <charly.chevalier@consensys.net>

darkwing commented on ui/pages/asset/util.ts:
```diff
@@ -64,3 +64,31 @@ export const chainSupportsPricing = (chainId: `0x${string}`) =>
 /** The opacity components should set during transition */
 export const loadingOpacity = 0.2;
+
+// TODO: Add unit tests
```
Once a few of these are in I'll approve!

gambinish replied:
Added unit test here: https://github.com/MetaMask/metamask-extension/pull/28429/commits/1a1737a7a6f215464bc6c70c163a039e972a7a0a

gambinish replied:
Updated with bugfix here: https://github.com/MetaMask/metamask-extension/pull/28429/commits/41adb561acf19278ba2a2f52fafa678d98f8e7d4

pedronfigueiredo committed:
fix: Default to dApp suggested fees only when user selects the option (#28403)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

Previously, some gas fee parameters defaulted to dApp suggested fee
values. This was problematic, because these values are always defined,
even when the user changed to a custom priority.

To fix the issue, we now only show the fees contained in txParams,
trusting they match the user selected priority level in the gas fees
modal.

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/28403?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

See screen recording below.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->



https://github.com/user-attachments/assets/3874d3b6-ad96-4fd9-9f40-361b38a0253b



### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

gambinish committed:
chore: Add new unit test for findAssetByAddress util

gambinish commented on ui/pages/asset/util.ts:
```diff
@@ -64,3 +64,31 @@ export const chainSupportsPricing = (chainId: `0x${string}`) =>
+  }
+
+  if (!address) {
+    return tokens.find((token) => !token.address);
```
Hmm that's a good point. This case may not be correct. Let me address this

gambinish replied:
There might be a case when the address is not provided, but we wouldn't want to grab the first matching token without an address in that case.

gambinish replied:
Addressed: https://github.com/MetaMask/metamask-extension/pull/28429/commits/41adb561acf19278ba2a2f52fafa678d98f8e7d4

gambinish committed:
fix: Update nullish search bug in chainSupprotsPricing

gambinish committed:
fix: Lint error in util.test.ts

gambinish committed:
Revert "fix: Lint error in util.test.ts"

This reverts commit aa3063f7a853ef6164165618394239cda87580ee.

gambinish committed:
Merge branch 'develop' into feat/mmassets-432_network-filter-extension--prep-updates

gambinish committed:
fix: @metamask/assets-controllers should be imported before local filepath

darkwing reviewed (approved):
Testing and all works great!

gambinish committed:
Merge branch 'develop' into feat/mmassets-432_network-filter-extension--prep-updates

gambinish committed:
Merge branch 'develop' into feat/mmassets-432_network-filter-extension--prep-updates
